Hotsapi.Uploader.Common/Uploader.cs

```diff
@@ -35,8 +37,9 @@ public Uploader()
         /// <param name="file"></param>
         public async Task Upload(ReplayFile file)
         {
-            file.UploadStatus = UploadStatus.InProgress;
-            if (file.Fingerprint != null && await CheckDuplicate(file.Fingerprint)) {
+            var checkDuplicate = file.UploadStatus != UploadStatus.ReadyForUpload;
+            file.UploadStatus = UploadStatus.Uploading;
+            if (file.Fingerprint != null && !checkDuplicate || await CheckDuplicate(file.Fingerprint)) {
```

Comment:
Could you wrap part of this conditional in parenthesis, for clarity? :) I don't know order of operations at a glance here

Is it

`(file.Fingerprint != null && !checkDuplicate) || await CheckDuplicate(file.Fingerprint)`
or
`file.Fingerprint != null && (!checkDuplicate || await CheckDuplicate(file.Fingerprint))`

Reply:
that's a bug